src/content/docs/en/guides/authentication.mdx

```diff
@@ -188,6 +188,124 @@ const session = await getSession(Astro.request);
 - [`auth-astro` on GitHub](https://github.com/nowaythatworked/auth-astro?tab=readme-ov-file#auth-astro)
 - [Auth.js documentation](https://authjs.dev/)
 
+## Better Auth
+
+Better Auth is a framework-agnostic authentication (and authorization) framework for TypeScript. It provides a comprehensive set of features out of the box and includes a plugin ecosystem that simplifies adding advanced functionalities.
+
+It supports Astro out of the box, and you can use it to add authentication to your astro project.
+
+### Installation
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install better-auth
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add better-auth
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add better-auth
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+For detailed setup instructions, check out the [Installtion Guide](https://www.better-auth.com/docs/installation).
+
+### Configuration
+
+Once you have installed and configured Better Auth in your project, you'll need to mount Better Auth handler in your Astro project.
+
+```ts title="page/api/auth/[...all].ts"
+import { auth } from "~/auth"; // import your Better Auth instance
+import type { APIRoute } from "astro";
+ 
+export const ALL: APIRoute = async (ctx) => {
+	return auth.handler(ctx.request);
+};
+```
+
+Follow the [Better Auth Astro Guide](https://www.better-auth.com/docs/integrations/astro) to learn more.
+
+### Usage
+
+Better Auth offers clients for various frameworks, including Vanilla JS, React, Vue, Svelte, and Solid. To create a client for your chosen framework:
+
+```ts title="src/lib/auth-client.ts"
+import { createAuthClient } from 'better-auth/client';
+
+export const authClient = createAuthClient({
+  // your configuration
+});
+
+export const { signIn, signOut } = authClient;
+```
+
+Once your client is set up, you can use it to authenticate users in your Astro components or any framework-specific files.
+
+
+```astro title="src/pages/index.astro"
+---
+import Layout from 'src/layouts/Base.astro';
+---
+<Layout>
+  <button id="login">Login</button>
+  <button id="logout">Logout</button>
+
+  <script>
+    const { signIn, signOut } = await import("./lib/auth-client")
+    document.querySelector("#login").onclick = () => signIn.social({
+      provider: "github",
+      callbackURL: "/dashboard",
+    })
+    document.querySelector("#logout").onclick = () => signOut()
+  </script>
+</Layout>
+```
+
+You can then use the `auth` object to get the user's session in your server-side code.
+
+```astro title="src/pages/index.astro"
+---
+import { auth } from "@/auth"; // import your Better Auth instance
+ 
+const session = await auth.api.getSession({
+	headers: Astro.request.headers,
+});
+---
+
+<p>{session.user?.name}</p>
+```
+
+And you can use the `auth` object to protect your routes as well.
```

Comment:
```suggestion
You can also use the `auth` object to protect your routes using middleware. The following example checks whether a user trying to access a logged-in dashbord route is authenticated, and redirects them to the home page if not.
```
A short description of what is happening in the code helps prep the reader to know what to look for!